vars/piperExecuteBin.groovy

```diff
@@ -0,0 +1,83 @@
+import com.sap.piper.JenkinsUtils
+import com.sap.piper.PiperGoUtils
+import com.sap.piper.Utils
+
+import static com.sap.piper.Prerequisites.checkScript
+
+void call(Map parameters = [:], stepName, metadataFile, List credentialInfo, failOnMissingReports = false, failOnMissingLinks = false) {
+
+    handlePipelineStepErrors(stepName: stepName, stepParameters: parameters) {
+
+        def script = checkScript(this, parameters) ?: this
+        parameters.script = null
```

Comment:
It is a common habit in this project to have the script reference as part of the parameters map. Can we be sure that the parameters map is not forwarded to somewhere where we again expect the script reference to be present? I understand that the script  - and other kind of instances - needs to be removed before the serialization towards go. But shouldn't we keep the parameters map itself unchanged?

Reply:
As we saw with some golang steps, it does not really harm to keep the script but I guess it is cleaner to get rid of the script here.

The provided step in this PR has the sole purpose to be the one and only thing to call when launching the binary.
Based on this assumption it should be okay to remove it from the map.

Reply:
I do not agree. Would be better not to modify the map we receive since we don't know if that map is used elsewhere (e.g. also handed over to another step (most probably not, but we can't make that decission here)) or we don't also know if the map is forwarded. But ok, in case you @OliverNocon  are sure that that all will not happen, go ahead.

Other point:

> The provided step

Is there any need for having that as a step (maybe one interesting question here is: do we expect that this step here is called directly from a pipeline). In case we don't expect this and this "step" here can be considered to be  some auxiliar coding it might be justified to have that as some kind of a util inside `src`.

Reply:
I just used a step due to simplicity reasons.
It will be for example simpler to test/mock in other steps I guess.

Reply:
> I just used a step due to simplicity reasons.

Not a good reason IMO